app/features/providers/github/commands/getUpcomingEvents.ts

```diff
@@ -8,21 +8,25 @@ import {
 } from '~/features/providers/misc/http'
 import type { Event } from './types'
 import { mapIssuesToEvents } from './misc/mapIssueToEvent'
+import { fetchCached } from './misc/cache'
 
 export const getUpcomingEvents = async (
   nextToken?: String
 ): Promise<ApiResponse<Event[]>> => {
   try {
     const client = newGraphQLClientFactory()
 
-    const res = await client<GetUpcomingEventsQuery>(getUpcomingEventsQuery, {
-      owner: 'cyprus-developer-community',
-      repo: 'events',
-      size: 20,
-      after: nextToken
-    })
+    const cacheKey = 'upcomingEvents'
+    const events = await fetchCached(cacheKey, async () => {
```

Comment:
I think the default ttl is a bit low in our case. Can you set it to 15 or 20 minutes?